users/views.py

```diff
@@ -88,3 +92,25 @@ def list(self, request):  # pylint:disable=unused-argument
         queryset = sorted(list(pycountry.countries), key=lambda country: country.name)
         serializer = CountrySerializer(queryset, many=True)
         return Response(serializer.data)
+
+
+class UsersViewSet(viewsets.ReadOnlyModelViewSet):
+    """Provides an API for listing system users. This is for the staff
+    dashboard."""
+
+    serializer_class = StaffDashboardUserSerializer
+    authentication_classes = [SessionAuthentication]
+    permission_classes = [IsAdminUser]
+    pagination_class = RefinePagination
+
+    def get_queryset(self):
+        name_search = self.request.query_params.get("q")
+
+        if name_search is None:
+            return User.objects.all()
+
+        return User.objects.filter(
+            Q(username__icontains=name_search)
+            | Q(name__icontains=name_search)
+            | Q(email__icontains=name_search)
+        ).all()
```

Comment:
We can use [SearchFilter](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter).